build/teamcity/util/roachtest_util.sh

```diff
@@ -19,7 +19,7 @@ stats_dir="$(date +"%Y%m%d")-${TC_BUILD_ID}"
 # Set up a function we'll invoke at the end.
 function upload_stats {
  if tc_release_branch; then
-      bucket="cockroach-nightly-${CLOUD}"
+      bucket="${ROACHTEST_BUCKET:-cockroach-nightly-${CLOUD}}"
```

Comment:
I missed this in my first review. Why is this bucket changing? Isn't this a net new bucket that only gets the workload replay data put in it? If there's any other output data, wouldn't that still go to the normal bucket? Am I missing something?

CC @rachitgsrivastava

Reply:
I thought the point was that the output data should not be in the normal public bucket.

Reply:
Not really much of a private test if all the output is there for everyone to see.

Reply:
Doh. Yes, you're right that the normal output needs to be put in a private bucket. But the `roachtest-snowflake-costfuzz` bucket is a bucket specifically for the workload replay data.

Looking at the `cockroach-nightly*` buckets, they are  set to `Subject to object ACLs` for what is public. So we do need a private bucket for that data. How about a new `crl-nightly-gcp-private` or `crl-nightly-gcp-internal` bucket?

Reply:
@rachitgsrivastava can you help with answering that question?

Reply:
This bucket name is actually [hard-coded](https://cockroachlabs.slack.com/archives/D01Q98X1JBZ/p1681757675426179). So the only thing that we need is the bucket where the actual results are stored.

Reply:
👍 

[Here's](https://github.com/cockroachdb/cockroach/pull/95390/files#diff-60bb95745a57a4d21b58f4d002d124ba024d274b6b31e29dcfada07feaaf57c8R22) the direct link to the code for anyone reading later.